code/datums/components/_component.dm

```diff
@@ -52,11 +52,14 @@
 /datum/component/New(list/raw_args)
 	parent = raw_args[1]
 	var/list/arguments = raw_args.Copy(2)
-	if(Initialize(arglist(arguments)) == COMPONENT_INCOMPATIBLE)
-		stack_trace("Incompatible [type] assigned to a [parent.type]! args: [json_encode(arguments)]")
-		qdel(src, TRUE, TRUE)
-		return
-
+	switch(Initialize(arglist(arguments)))
+		if (COMPONENT_INCOMPATIBLE)
+			stack_trace("Incompatible [type] assigned to a [parent.type]! args: [json_encode(arguments)]")
+			qdel(src, TRUE, TRUE)
+			return
+		if(INITIALIZE_HINT_QDEL)
+			qdel(src, TRUE, TRUE)
+			return
```

Comment:
Hmmm this smells like #76005 a little bit. Not sure we should be doing this.

Reply:
I guess I can just repeatedly check for the global flag instead.